.github/scripts/Extensions.kts

```diff
@@ -0,0 +1,17 @@
+import java.io.File
+
+fun File.isGitDirty(): Boolean {
+  val gitDiffExitValue = runCommand("git diff --quiet").run {
+    exitValue()
+  }
+  return gitDiffExitValue == 1
+}
+
+fun File.runCommand(command: String): Process {
+  val parts = command.split("\\s".toRegex())
+
+  return ProcessBuilder(*parts.toTypedArray())
+      .directory(this)
+      .start()
+      .apply { waitFor() }
+}
```

Comment:
I don't think these should be extension functions on `java.io.File`, because they are not applicable to all `File` instances, only specific ones. I feel these should just be top level functions instead which accept a `File` as a parameter. What do you think?

Reply:
I initially went with that but had to pass the working directory in every function. So I just went with extensions, we can move these extensions in the `CommitAndPush.main.kts` file so that they are scoped to that script. WDYT?

Reply:
Yeah, that works